packages/starlight/utils/translations.ts

```diff
@@ -3,16 +3,23 @@ import config from 'virtual:starlight/user-config';
 import type { i18nSchemaOutput } from '../schemas/i18n';
 import { createTranslationSystem } from './createTranslationSystem';
 
-/** All translation data from the i18n collection, keyed by `id`, which matches locale. */
-let userTranslations: Record<string, i18nSchemaOutput> = {};
-try {
-	// Load the user’s i18n collection and ignore the error if it doesn’t exist.
-	userTranslations = Object.fromEntries(
-		// @ts-ignore — may be an error in projects without an i18n collection
-
-		(await getCollection('i18n')).map(({ id, data }) => [id, data] as const)
-	);
-} catch {}
+/** Get all translation data from the i18n collection, keyed by `id`, which matches locale. */
+async function loadTranslations() {
+	let userTranslations: Record<string, i18nSchemaOutput> = {};
+	// Briefly override `console.warn()` to silence logging when a project has no i18n collection.
+	const warn = console.warn;
+	console.warn = () => {};
```

Comment:
😞 This obviously works, but I'm wondering if we can make the warnings for `getCollection` lazy in core? Ideally you'd only get the warning if you try to _access_ a collection that does not exist.

Reply:
That’s exactly what’s happening here! We call `getCollection('i18n')` _in case_ it exists (there’s no `hasCollection('i18n')` of `getCollections()` API we could use to check for existence first).